File: DemoServer/Indexes/ProductName.cs
using System.Linq;
using DemoServer.Entities;
using Raven.Client.Documents.Indexes;

namespace DemoServer.Indexes
{
    public class ProductName : AbstractIndexCreationTask<Product>
    {
        public ProductName()
        {
            Map = products => from product in products
                              select new
                              {
                                  product.Name
                              };
        }
    }
}

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/indexes/ProductName.java
package net.ravendb.demo.indexes;

import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;

public class ProductName extends AbstractIndexCreationTask {

    public ProductName() {
        map = "from product in docs.Products " +
                "select new { " +
                "  Name = product.Name " +
                "}";
    }

}
